pkg/controller/machine/controller.go

```diff
@@ -383,6 +399,22 @@ func (r *ReconcileMachine) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	// At this point, we are certain that the Machine we have in memory has not been created already,
+	// so we are going to ask the actuator to create it.
+	// To avoid potential eventual consistency errors from having a stale informer cache, make sure to
+	// get the latest machine directly from the Kube API to send into the actuator.
+	directM := &machinev1.Machine{}
+	if err := r.directClient.Get(ctx, request.NamespacedName, directM); err != nil {
+		return reconcile.Result{}, fmt.Errorf("could not fetch machine with direct client: %w", err)
+	}
+	if m.GetResourceVersion() != directM.GetResourceVersion() {
+		// The object we retrieved from the informer is not up to date.
+		// We should not attempt to create a new machine with a stale object.
+		// Requeue until the informer has the latest version.
+		klog.V(2).Infof("%v: resource versions mismatch, reconciling stale object, requeue (%s vs %s)", m.GetName(), m.GetResourceVersion(), directM.GetResourceVersion())
+		return reconcile.Result{Requeue: true}, nil
```

Comment:
As mentioned in person, I'd add a slight `RequeueAfter` delay here to avoid re-queueing immediately to give some time for the change to propagate through to the informer watching the object.

Reply:
+1